docs/Style.thy

```diff
@@ -248,6 +249,38 @@ text \<open>
   not what it is doing.\<close>
 
 
+section \<open>Fun, primrec, or case? case!\<close>
+
+text \<open>
+  When defining a non-recursive function that uses pattern matching, there are multiple options:
+  @{command fun}, @{command primrec}, and @{text case}. @{text case} initially seems like the least
+  powerful of these, but combined with the @{attribute split_simps} attribute, it is the option that
+  provides the highest degree of proof automation. @{command fun} and @{command primrec} provide
+  simp rule for each of the pattern cases, but when a closed term appears in the proof, there is no
+  way to automatically perform a case distinction -- you need to do a manual @{method cases} or
+  @{method case_tac} invocation. @{text case} does provide automatic splitting via split rules, and
+  @{attribute split_simps} adds the missing simp rules.
+
+  So the recommended pattern is to use @{text case} in the definition and directly after, via
+  attribute, add the simp rules to the global simp set as @{command fun}/@{command primrec} would
```

Comment:
Sentence is backwards, confusing. Suggest something like:
"The recommended pattern is to use @{text case} in the definition and supply the simp rules directly after it, adding them to the global simp set as ... would do."